ctapipe/io/tableio.py

```diff
@@ -350,26 +350,106 @@ class FixedPointColumnTransform(ColumnTransform):
 
     Can be used to store values as fixed point by using an integer dtype
     and a scale that is a power of 10.
+
+    The transforms reserves 3 integers to represent -inf, nan and inf.
+    Underflowing values are converted to -inf and overflowing to inf.
+
+    For unsigned target dtype:
+    -inf: maxval - 2
+    nan: maxval - 1
+    inf: maxval
+
+    For signed target dtype:
+    -inf: minval
+    nan: minval + 1
+    inf: maxval
+
+    When reading, these special values must not be interpreted as valid integer
+    values but be transformed back into -inf, nan, inf respectively.
+
+    This is a lossy transformation.
     """
 
     def __init__(self, scale, offset, source_dtype, target_dtype):
         self.scale = scale
         self.offset = offset
         self.source_dtype = np.dtype(source_dtype)
         self.target_dtype = np.dtype(target_dtype)
+        self.unsigned = self.target_dtype.kind == "u"
+
+        iinfo = np.iinfo(self.target_dtype)
+
+        # use three highest values for nan markers for unsigned case
+        if self.unsigned:
+            self.neginf = iinfo.max - 2
+            self.nan = iinfo.max - 1
+            self.posinf = iinfo.max
+
+            # this leaves this inclusive range for the valid values
+            self.minval = 0
+            self.maxval = iinfo.max - 3
+        else:
+            self.neginf = iinfo.min
+            self.nan = iinfo.min + 1
+            self.posinf = iinfo.max
+
+            # this leaves this inclusive range for the valid values
+            self.minval = iinfo.min + 2
+            self.maxval = iinfo.max - 1
 
     def __call__(self, value):
-        return (value * self.scale).astype(self.target_dtype) + self.offset
+        is_scalar = np.array(value, copy=False).shape == ()
```

Comment:
Numpy docs suggests `np.ndim(value) == 0`
https://numpy.org/doc/stable/reference/generated/numpy.isscalar.html

Reply:
This is what is done in astropy. There probably was a problem with `isscalar` and they didn't check maybe?

Reply:
`np.isscalar` wont work on 0-dim arrays, `np.ndim` (and this implementation) does.

> If you need a stricter way to identify a numerical scalar, use isinstance(x, numbers.Number), as that returns False for most non-numerical elements such as strings.
In most cases np.ndim(x) == 0 should be used instead of this function, as that will also return true for 0d arrays. This is how numpy overloads functions in the style of the dx arguments to gradient and the bins argument to histogram.

This is probably fine as well. I was just wondering where it comes from

Reply:
I fail to see a reason why this would not work so I assume it covers all the relevant cases